### PR TITLE
Docs issues

### LINKED ISSUE #657: Update `net.models` documentation
### Feature Request

It is strange and a little bit outdated (I don't like it personally). Some transaction classes are missing.

Should be done after #649

### LINKED ISSUE #773: Investigate `Transaction too big to fit a batch` error
### Feature Request

https://testnet-2.starkscan.co/tx/0x20440925a18ba8911f2fe2bbbcb64511ca5f3d7bffaa036ea3eda0f9cef26f6

Using a multicall with too many calls results in the above error (supposedly one more similiar error is possible).

### LINKED ISSUE #909: Add a note to docs about starknet project services
### Feature Request

It could be in the footnote for example.

### LINKED ISSUE #917: Document omitted modules
### Feature Request

Some modules are not in the documentation, despite them having docstrings.

One example would be `starknet_py.common`

### PR DESCRIPTION
**Pull request by tomek0123456789**
Closes #917 (not sure about it though)
Closes #909
Closes #773
Closes #657

## Introduced changes

- Changed docs as requested in issues mentioned above

##

- [x] This PR contains breaking changes

- Changed file name from `transaction_exceptions.py` to `transaction_errors.py` (to match other files)

### CODE REVIEW

File: docs/_templates/page.html [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    {%- if last_updated -%}
+    <div class="last-updated">
+        {% trans last_updated=last_updated|e -%}
+        Last updated on {{ last_updated }}
+        {%- endtrans -%}
+    </div>
+    {%- endif %}
+    </div>
+</div>
+{% endblock %}
```
Comment (drknzz) [inline docs/_templates/page.html:68]: Let's add a newline 👀

File: starknet_py/net/models/transaction.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def type(self) -> TransactionType:
     def calculate_hash(self, chain_id: StarknetChainId) -> int:
         """
         Calculates the transaction hash in the Starknet network - a unique identifier of the
-        transaction. See :ref:`compute_transaction_hash` docstring for more details.
+        transaction. See :py:meth:`~starknet_py.hash.transaction.compute_transaction_hash` docstring for more details.
```
Comment (drknzz) [inline starknet_py/net/models/transaction.py:61]: 👍

**Review by drknzz — COMMENTED** [review]
Nice job overall ⭐

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -5,7 +5,7 @@ Migration guide
 0.16.0 Migration guide
 **********************
 
-    Version 0.16.0 of **starknet.py** comes with *support for Python 3.8, 3.9, 3.10 and 3.11* !
+    Version 0.16.0 of **starknet.py** comes with *support for Python 3.8, 3.9, 3.10 and 3.11*!
```
[the remark this answers is not in the document]
Comment (tomek0123456789) [reply docs/migration_guide.rst:8]: I guess we need a council meeting

cc @war-in

File: docs/api/models.rst [hunk 4]
```diff
@@ -5,10 +5,22 @@ Module containing base models and functions to operate on them.
...
+    :members:
+
+
+.. autoclass:: DeployAccount
+
+.. autoclass:: Declare
+
+.. autoclass:: DeclareV2
+
+.. autoclass:: Invoke
```
[the remark this answers is not in the document]
Comment (tomek0123456789) [reply docs/api/models.rst:23]: There aren't any as far as I can see, will remove them, thanks!

File: docs/migration_guide.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (war-in) [reply docs/migration_guide.rst:8]: In my opinion, space is unnecessary. We also have external contribution addressing this issue https://github.com/software-mansion/starknet.py/pull/1009 😇

**Review by war-in — APPROVED** [review]

**Pull request merged**